apps/evm/src/lib/swap/useCrossChainTrade/useSquidCrossChainTrade.ts

```diff
@@ -0,0 +1,402 @@
+import { Percent, ZERO_PERCENT } from 'sushi/math'
+import { RouterLiquiditySource } from 'sushi/router'
+import { RToken } from 'sushi/tines'
+import { encodeFunctionData, stringify } from 'viem'
+import {
+  TransactionType,
+  decodeSquidRouterCallData,
+  encodeSquidBridgeParams,
+  encodeSwapData,
+  getBridgeParams,
+  getSquidRouteRequest,
+  isSquidRouteProcessorEnabled,
+} from './SushiXSwap2'
+import { UseCrossChainTradeParams, UseCrossChainTradeReturn } from './types'
+import { useSquidRoute } from './useSquidRoute'
+
+export const useSquidCrossChainTrade = ({
+  network0,
+  network1,
+  token0,
+  token1,
+  amount,
+  slippagePercentage,
+  recipient,
+  enabled,
+  tradeId,
+}: Omit<UseCrossChainTradeParams, 'network0' | 'network1'> & {
+  network0: SquidAdapterChainId
+  network1: SquidAdapterChainId
+}) => {
+  const { data: gasPrice0 } = useGasPrice({
+    chainId: network0,
+    query: { enabled },
+  })
+  const { data: gasPrice1 } = useGasPrice({
+    chainId: network1,
+    query: { enabled },
+  })
+
+  const { address } = useAccount()
+
+  const { bridgePath, isSrcSwap, isDstSwap } = useMemo(() => {
+    const bridgePath =
+      isSquidAdapterChainId(network0) && isSquidAdapterChainId(network1)
+        ? {
+            srcBridgeToken: axlUSDC[network0],
+            dstBridgeToken: axlUSDC[network1],
+          }
+        : undefined
+
+    const isSrcSwap = Boolean(
+      token0 &&
+        bridgePath?.srcBridgeToken &&
+        !token0.equals(bridgePath.srcBridgeToken),
+    )
+
+    const isDstSwap = Boolean(
+      token1 &&
+        bridgePath?.dstBridgeToken &&
+        !token1.equals(bridgePath.dstBridgeToken),
+    )
+
+    return {
+      bridgePath,
+      isSrcSwap,
+      isDstSwap,
+    }
+  }, [network0, network1, token0, token1])
+
+  const { data: srcTrade } = useApiTrade({
+    chainId: network0,
+    fromToken: token0,
+    toToken: bridgePath?.srcBridgeToken,
+    amount,
+    slippagePercentage,
+    gasPrice: gasPrice0,
+    recipient: SQUID_ROUTER_ADDRESS[network0],
+    enabled:
+      enabled &&
+      isSrcSwap &&
+      isSquidRouteProcessorEnabled[network0] &&
+      Boolean(amount),
+    carbonOffset: false,
+    source: RouterLiquiditySource.XSwap,
+    onError: () => {
+      log.error('xswap src swap api error')
+    },
+  })
+
+  const { data: dstTrade } = useApiTrade({
+    chainId: network1,
+    amount: isSrcSwap ? srcTrade?.minAmountOut : amount,
+    fromToken: bridgePath?.dstBridgeToken,
+    toToken: token1,
+    slippagePercentage,
+    gasPrice: gasPrice1,
+    recipient,
+    enabled:
+      enabled &&
+      isDstSwap &&
+      isSquidRouteProcessorEnabled[network1] &&
+      Boolean(isSrcSwap ? srcTrade?.minAmountOut : amount),
+    carbonOffset: false,
+    source: RouterLiquiditySource.XSwap,
+    onError: () => {
+      log.error('xswap dst swap api error')
+    },
+  })
+
+  const squidRouteRequest = useMemo(
```

Comment:
I personally prefer full fns in things like memos, it's more readable. Unless it's a one liner ig.

Makes things like debugging later on a tiny bit more pleasant (can't directly place a console.log in here)